src/databricks/labs/ucx/workspace_access/groups.py

```diff
@@ -624,6 +624,46 @@ def delete_original_workspace_groups(self):
         # Step 3: Confirm that enumeration no longer returns the deleted groups.
         self._wait_for_deleted_workspace_groups(deleted_groups)
 
+    def pick_owner_group(self, prompt: Prompts, username: str) -> str | None:
+        # This method is used to select the group that will be used as the owner group.
+        # The owner group will be assigned by default to all migrated tables/schemas
+        groups = self._user_account_groups(username)
+        if not groups:
+            logger.warning("No account groups found for the current user.")
+            return None
+        if len(groups) == 1:
+            return groups[0].display_name
+        group_names = [group.display_name for group in groups]
+        return prompt.choice("Select the group to be used as the owner group", group_names, max_attempts=3)
+
+    def validate_owner_group(self, group_name: str) -> bool:
+        # This method is used to validate that the current owner is a member of the group
+        username = self._ws.current_user.me().user_name
+        if not username:
+            logger.warning("No user found for the current session.")
+            return False
+        groups = self._user_account_groups(username)
+        if not groups:
+            logger.warning("No account groups found for the current user.")
+            return False
+        group_names = [group.display_name for group in groups]
+        return group_name in group_names
+
+    def _user_account_groups(self, username: str) -> list[Group]:
+        # This method is used to find all the account groups that a user is a member of.
+        groups: list[Group] = []
+        account_groups = self._list_account_groups("id,displayName,externalId,members")
+        if not account_groups:
+            return groups
+        for group in account_groups:
+            if not group.members:
+                continue
+            for member in group.members:
+                if member.display == username:
```

Comment:
does it work with service principal?

Reply:
Nope! And it doesn't work with users too ;-)
Fixed